### PR TITLE
Reactivate big pixels for the Phase 1 pixel topology

### PR DESCRIPTION
**Pull request by jkarancs**
These changes enable big pixels along the edges of the pixel readout chips.
They were disabled in the Phase 1 TDR, but the actual sensor design is the same as that of the Phase 0 detector so the relevant parameters were reverted in this PR.

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
@franzoni - FYI, this change will require a GT update

**Comment by VinInn** [comment]
@rovere @boudoul @veszpv @slava77 
I am sure you are all very interested in this...

**Comment by jkarancs** [comment]
Additional people of interest to watch this PR: @veszpv @tsusa @dkotlins @pmaksim1

**Comment by franzoni** [comment]
hello @jkarancs,
are the 'big pixels' sensitive volumes, I guess ?
Are they described somewhere, so we can gauge how far this change will propagate.
@ghellwig @arunhep may want to follow this too.

**Comment by jkarancs** [comment]
Hello @franzoni

Yes, they are sensitive volumes. In my preliminary studies with sim to rechit comparison the previously observed biases in the template hit reconstruction for the BPix are no longer there, the difference for the generic RECO is negligible (only statistical), because it uses the PixelTopology class for the pixel size already. However templates have built in pixel size (100 um x 150 um, same as Phase 0) which created some bias earlier without big pixels (slide 4 [1]). The ROC size is fixed at 8.1mm, but without big pixels, the pixel size artificially changed to 101.25 um x 155.8 um (x and y). The observed biases (in particular in the local y direction) agrees with the change in pixel size. There will be improvement already with the current templates (but those will be updated too).

So to summarize: this PR will fix the position biases observed in the barrel.
Another PR will fix the template reco software for the FPix.

[1] - https://indico.cern.ch/event/536891/contributions/2376928/attachments/1374299/2086018/2016_11_18_PixelOfflineMeeting_PhaseIOfflinePlansUpdate.pdf

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
@jkarancs - just to clarify: is this PR the last with respect to Tracker geometry changes?

**Comment by jkarancs** [comment]
Dear @ianna,
I'd rather let @boudoul and @veszpv answer that question, because I'm not sure. At least from the pixel offline part, there are no more changes expected.

**Comment by veszpv** [comment]
@ianna -- at the moment, I am not aware of any further outstanding items that would need to go into pre4 which would result in any changes in the geometry description. (Only validation codes should be improved once manpower is available.)

@franzoni -- Janos explained the meaning of this PR: it fixes local reco with the present CPE-s. On the top of this, we expect simultaneous code change in CPE and new template DB tags preferably in pre4, which would improve performance.

**Comment by ianna** [comment]
@veszpv - thanks! I'll prepare the new payloads later today. If nobody objects I'd like also to add Tracker Overlap Bugfixes #17383 and Replace Polycone in Pixel Fwd #17387

**Comment by ianna** [comment]
@franzoni and @jkarancs - FYI, the tracker reco geometry does not change. There are only two records which are affected GeometryFileRcd and PTrackerParametersRcd. 

Here are the tags for upgrade2017 GT queues:
```
XMLFILE_Geometry_90YV6_Extended2017_mc
TKParameters_Geometry_90YV6
```
and for upgrade2018 GT queues:
```
XMLFILE_Geometry_90YV7_Extended2018_mc
TKParameters_Geometry_90YV6
```

**Comment by ghellwig** [comment]
@ianna just for my education:
What exactly is the purpose of the individual records? Typically, you update one or more of the following ones:
 * `GeometryFileRcd`
 * `PTrackerParametersRcd`
 * `IdealGeometryRecord`
 * `PGeometricDetExtraRcd`

From [your comment](https://github.com/cms-sw/cmssw/pull/17384#issuecomment-277209278) it seems to me that the only record that would trigger an update of the tracker alignment payload is `IdealGeometryRecord`. Is that correct?

**Comment by venturia** [comment]
@ghellwig 's question is tricker than it sounds. If you are talking about an ideal MC alignment, yes, likely IdealGeometryRecord should be the only one which affects the alignment payload. But, for example, the change of the big pixels which is fixed by Ianna's change, can affect the measurement of the alignment, and therefore the alignment payload which is produced.

**Comment by ianna** [comment]
@gartung - 

* GeometryFileRcd is a geometry scenario - it includes consistent detector description including some reco and sim parameters. It corresponds to one compressed xml file created from a file list defined here: http://cmslxr.fnal.gov/source/Geometry/CMSCommonData/python/cmsExtendedGeometry2017XML_cfi.py
* PTrackerParametersRcd is a list of tracker parameters defined here: http://cmslxr.fnal.gov/source/Geometry/TrackerCommonData/data/PhaseI/trackerParameters.xml
and used in TrackerTopology: https://github.com/cms-sw/cmssw/tree/CMSSW_9_0_X/Geometry/TrackerNumberingBuilder
* IdealGeometryRecord is a Tracker reco geometry, e.g. sensitive volumes only (DetUnits)
* PGeometricDetExtraRcd is Tracker Extra information http://cmslxr.fnal.gov/source/Geometry/TrackerNumberingBuilder/interface/GeometricDetExtra.h

I did not look at the alignment code, so I cannot tell you if you rely only on IdealGeometryRecord.

**Comment by gartung** [comment]
@ianna I think you meant @ghellwig

**Comment by ghellwig** [comment]
Thank you for the clarifying comments @ianna and @venturia!
I was a bit unspecific in my question, but @venturia was guessing in the right direction.
I was interested in the record which needs to kept in synch with the ideal MC alignment. This is the record which needs to monitored from the alignment side to have the ideal MC alignment and the tracker geometry consistent.

**Comment by mmusich** [comment]
Hello @ghellwig this PR has two implications:
  * since `IdealGeometryRecord` is not updated, there is no need for an update of the alignment record in the 2017/2018 scenarios global tag (as long as the Pixel alignment in those scenarios is still kept to ideal)
  * when deriving a realistic scenario based on an *actual* alignment done with tracks (i.e. minimizing a chi2 and so on), one better use as starting point samples reconstructed with this feature, not to include inconsistencies later on.

**Comment by ghellwig** [comment]
>  * when deriving a realistic scenario based on an actual alignment done with tracks (i.e. minimizing a chi2 and so on), one better use as starting point samples reconstructed with this feature, not to include inconsistencies later on.

@mmusich that is true, but my question was really just targeting your first point

**Comment by franzoni** [comment]
@ianna
I could provide you w/ a GT which includes the tags of above
in the form of a cherry pick for you
Or you can get this PR separately (w/ PPD workshop ongoing, only limited time till tomorrow evening) merged, and I'll deploy the two new tags in a subsequent GT update

**Comment by boudoul** [comment]
Hi @franzoni , I fear @ianna is away these days ....

**Comment by ianna** [comment]
@franzoni - thanks. As discussed at Simulation meeting on Friday these are the last changes to 2017 geometry. It's declared "frozen", so there is no rush

**Comment by franzoni** [comment]
hello @ianna 
please see https://github.com/cms-sw/cmssw/pull/17437 
where I've addressed https://github.com/cms-sw/cmssw/pull/17384#issuecomment-277209278
( together with two other changes, there documented )

If you / @slava77 deem useful cherry-picking my commit into this PR, feel free to do so
(since you might be out-of-office as @boudoul suggests, I've placed a PR)

**Comment by ianna** [comment]
@franzoni - thanks! I think the PR integration order should pose no problems.

**Comment by davidlange6** [comment]
+1
